### PR TITLE
Adds special handling and logger when the API response is malformed

### PR DESCRIPTION
**Pull request by hsume2**
@hashwin, this adds special logging for the Airbrakes that we are seeing.

### CODE REVIEW

File: lib/twitter/rest/utils.rb [hunk 1]
```diff
@@ -93,7 +93,17 @@ def perform_post_with_objects(path, options, klass)
...
       # @param klass [Class]
       def perform_request_with_objects(request_method, path, options, klass)
-        perform_request(request_method, path, options).collect do |element|
+        result = perform_request(request_method, path, options)
+        unless result.respond_to?(:collect)
+          @logger ||= begin
+            require 'logger'
+            ::Logger.new(STDOUT)
+          end
+          @logger.info "[#{self.class.name}##{__method__}] #{request_method}, #{path}, #{result.inspect}"
```
Comment (hashwin) [inline lib/twitter/rest/utils.rb]: Should we also log the options?

File: lib/twitter/rest/utils.rb [hunk 2]
```diff
@@ -93,7 +93,17 @@ def perform_post_with_objects(path, options, klass)
       # @param options [Hash]
       # @param klass [Class]
       def perform_request_with_objects(request_method, path, options, klass)
-        perform_request(request_method, path, options).collect do |element|
+        result = perform_request(request_method, path, options)
```
Comment (hashwin) [inline lib/twitter/rest/utils.rb:96]: I think we should have additional logging inside this method as well, to get a clearer idea of what exactly the response was as soon as we got it without the twitter gem playing around with it.

**Comment by hsume2** [comment]
@hashwin :haircut:

**Comment by hsume2** [comment]
@hashwin bump

**Comment by hashwin** [comment]
Specs failing?

**Comment by hsume2** [comment]
@hashwin :haircut:

**Comment by hashwin** [comment]
:+1:

**Pull request closed without merging**